packages/encryption/src/lib/encryption.ts

```diff
@@ -182,49 +135,39 @@ export const decryptFromIpfs = async (
       errorCode: LIT_ERROR.INVALID_PARAM_TYPE.name,
     });
 
-  try {
-    const metadata: EncryptToIpfsPayload = await (
-      await fetch(`https://gateway.pinata.cloud/ipfs/${ipfsCid}`)
-    ).json();
-    if (metadata.dataType === 'string') {
-      return decryptToString(
-        {
-          accessControlConditions: metadata.accessControlConditions,
-          evmContractConditions: metadata.evmContractConditions,
-          solRpcConditions: metadata.solRpcConditions,
-          unifiedAccessControlConditions:
-            metadata.unifiedAccessControlConditions,
-          ciphertext: metadata.ciphertext,
-          dataToEncryptHash: metadata.dataToEncryptHash,
-          chain: metadata.chain,
-          authSig,
-          sessionSigs,
-        },
-        litNodeClient
-      );
-    } else {
-      return decryptToFile(
-        {
-          accessControlConditions: metadata.accessControlConditions,
-          evmContractConditions: metadata.evmContractConditions,
-          solRpcConditions: metadata.solRpcConditions,
-          unifiedAccessControlConditions:
-            metadata.unifiedAccessControlConditions,
-          ciphertext: metadata.ciphertext,
-          dataToEncryptHash: metadata.dataToEncryptHash,
-          chain: metadata.chain,
-          authSig,
-          sessionSigs,
-        },
-        litNodeClient
-      );
-    }
-  } catch (e) {
-    return throwError({
-      message: 'Unable to fetch or decrypt from IPFS',
-      errorKind: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.kind,
-      errorCode: LIT_ERROR.INVALID_ARGUMENT_EXCEPTION.name,
-    });
+  // FIXME: The return type of this function is inferrable based on the value of `params.dataType`
+  if (parsedJsonData.dataType === 'string') {
+    return decryptToString(
+      {
+        accessControlConditions: parsedJsonData.accessControlConditions,
+        evmContractConditions: parsedJsonData.evmContractConditions,
+        solRpcConditions: parsedJsonData.solRpcConditions,
+        unifiedAccessControlConditions:
+          parsedJsonData.unifiedAccessControlConditions,
+        ciphertext: parsedJsonData.ciphertext,
+        dataToEncryptHash: parsedJsonData.dataToEncryptHash,
+        chain: parsedJsonData.chain,
+        authSig,
+        sessionSigs,
+      },
+      litNodeClient
+    );
+  } else {
+    return decryptToFile(
```

Comment:
Do we always want to default to the file?

Reply:
The `dataType` currently can only be `string` or `file` (it's a string union type), so currently the only possible option if `dataType` is not `string` is that it must be `file` :)

Reply:
yes, it might be worth adding an explicit check on the `dataType` so if in the future we add more to the type union it will not default if someone doesn't update this conditional check.

Reply:
Roger that :+1: :) -- I updated this to make the check explicit and to explicitly throw an error if the value isn't `string` or `file` -- note that we're also validating it in the `decryptFromJson` params validator, so someone will also need to modify that validator to support the new string before this `else{}` logic will throw the new exception.